hack/update-deps.sh

```diff
@@ -1,25 +1,4 @@
-#!/bin/bash -e
-
-readonly GLIDE_MINOR_VERSION="13"
-readonly REQUIRED_GLIDE_VERSION="0.$GLIDE_MINOR_VERSION"
-
-function verify_glide_version() {
-	if ! command -v glide &> /dev/null; then
-		echo "[FATAL] Glide was not found in \$PATH. Please install version ${REQUIRED_GLIDE_VERSION} or newer."
-		exit 1
-	fi
-
-	local glide_version
-	glide_version=($(glide --version))
-	if ! echo "${glide_version[2]#v}" | awk -F. -v min=$GLIDE_MINOR_VERSION '{ exit $2 < min }'; then
-		echo "Detected glide version: ${glide_version[*]}."
-		echo "Please install Glide version ${REQUIRED_GLIDE_VERSION} or newer."
-		exit 1
-	fi
-}
-
-verify_glide_version
-
-glide update --strip-vendor
-
-find vendor/ -name *.proto | xargs sed -i '/k8s.io\/apiextensions-apiserver\/pkg\/apis\/apiextensions\/v1beta1\/generated.proto/d'
+#!/bin/bash -eu
+go get -u=patch -d ./...
+go mod tidy
+go mod vendor
```

Comment:
is that script new? Have not seen it elsewhere. Is it what we want what makes sense? cc @tnozicka

Reply:
This mostly covers the flow I used and described in https://github.com/openshift/enhancements/blob/master/enhancements/oc/upgrade_kubernetes.md. With one exception, I usually bumped only a specific dep.

Reply:
why is it not in the build system Makefiles?

Reply:
I think it's about time to do it now.

Reply:
Oh, I already migrated to 1.13 and didn't notice that `-u=patch` was introduced by go1.13. Also it doesn't require `GO111MODULE` to be set.

Any ideas how it can be updated for go1.12? `-u=true` might be too much.

Reply:
Since we're switching to go 1.13 globally, we'll just wait until that happens and we can do the switch w/o looking back.

Reply:
I'd prefer just to drop the script.

go mod can bump single dependency (+transitives) so I don't think this would be a flow to encourage, for backports smaller surface means safer

I think go get doesn't update replace directives.

Reply:
We'll need clear instruction to run tidy and vendor, otherwise ppl will forget about tidy, for sure.

Reply:
not sure patch is the universal option either, I am inclined not to codify the path

Reply:
(`go mod tidy` and `go mod vendor` will be enforced by CI though)

Reply:
> otherwise ppl will forget about tidy, for sure.
CI is the only way to enforce those, I'll try to speed up that

Reply:
/hold
I just confirmed that go get doesn't update a dependency with replace directive

Reply:
But replace directives are being pinned to a branch which get replaced with a tag to we endup with something fixed. The issue is that go mod is unable to track branches and assuming it makes word better by forcing semver tags is just faulty. Maybe we shouldn't use replace directives for that and just have a script with a set of `go get @master`. But until we sort it out on some arch call I'd prefer not to codify conflicting approaches.

Reply:
@mfojtik has goodmod.

Let's not solve this problem here in this PR, and let's remove the script for now.

Reply:
> Let's not solve this problem here in this PR, and let's remove the script for now.

Yeah, it looks like even though people use these steps, they are not hard requirements. @dmage mind dropping those bits from this PR?

Reply:
Dropped.